src/components/Video/Video.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {
+  useState,
+  useEffect,
+  useRef
+} from 'react';
+import styles from './Video.module.scss';
+import { VideoData } from '../../types';
+import LazyLoad from '../LazyLoad/LazyLoad';
+
+import videojs from 'video.js';
+import 'video.js/dist/video-js.css';
+
+export type VideoProps = {
+  hasControls: boolean,
+  isAutoplaying: boolean
+  isLooping: boolean,
+  isMuted: boolean,
+  isPlaying?: boolean,
+  video: VideoData,
+};
+
+const Video = ({ isAutoplaying = true, hasControls = true, isPlaying = true, isLooping = true, isMuted = true, video }: VideoProps) => {
+  const videoRef = useRef(null);
+  const playerRef = useRef(null);
+  const isFirstLoad = useRef(true);
+
+  const [hasLoaded, setHasLoaded] = useState(false);
+  const [isMounted, setIsMounted] = useState(false);
+
+
+  if (!video) {
+    return <div className={`${styles.frame} ${styles.brokenVideo}`} />
+  }
+
+
+  const onPlayerReady = (player) => {
+
+    // TODO: Add custom buttons on the player (download)
+
+    // let downloadButtonElement = document.createElement('button')
+    // downloadButtonElement.innerHTML = '⤓';
+
+    // console.log(player.controlBar);
+
+    // player.controlBar.el().insertBefore(downloadButtonElement, player.controlBar.getChild('FullscreenToggle').el());
+
+    // player.addClass('vjs-vjsdownload');
+  };
+
+  useEffect(() => {
+
+    if (isFirstLoad.current && videoRef.current && video) {
+      isFirstLoad.current = false;
+      const videoElement = document.createElement("video-js");
+
+      videoElement.classList.add('vjs-big-play-centered');
+      videoRef.current.appendChild(videoElement);
+
+      const player = playerRef.current = videojs(videoElement, {
+        sources: [{
+          src: video.hls,
+          type: "application/x-mpegURL"
+        }],
+        autoplay: isAutoplaying,
+        muted: isMuted,
+        controls: hasControls,
+        fluid: true,
+        controlBar: {
+          pictureInPictureToggle: false, //firefox
```

Comment:
<img width="133" alt="image" src="https://github.com/RandomStudio/random-studio/assets/5897209/26eff218-8625-4080-829b-b399837190ea">
This doesn't seem to have worked